### ISSUE TITLE
Cast the Typescript type for a String column for non-DB enums

### ISSUE THREAD

**Issue by mmahalwy** [prisma/prisma#7081]
## Problem
Is there a way to use a `String` column as an enum without creating the enum type on the db? I got an often-changing string column that id like to limit to a few values and have those values be casted as Typescript types. 

For example:
```ts
enum FormFieldType {
  Input
  Textarea
  Select
  ...
}
```

```ts:schema.prisma
model FormField {
name String
type  String  @cast(FormFieldType)
}
```

This would cast a TS type (notice `type` key not a `string`): 
```ts
type FormField = {
  name: string;
  type: FormFieldType;
}
```

This is also great when creating or updating the model where `type` column has an enum type, disallowing non-allowed values.

## Suggested solution
Suggestion noted above. Allow a directive for the schema to specify the cast type. Another option is to override this in the options when initializing the `PrismaClient` class.

## Alternatives
Noted above.

**Comment by janpio** [comment]
Prisma does not support this functionality right now. A string is a string, and an enum is an enum. You could probably prevent adding unwanted values via a simple middleware, but that would not give you the type safety you are suggesting above unfortunately.

**Comment by mmahalwy** [comment]
@janpio Agree, a middleware could be an option but will not solve the type-safety concerns. While I agree that a string is a string, database column types don't always translate well when it comes to typescript types (string literals, as an example, or number vs bigint/float). 

We could solve this issue and this [issue](https://github.com/prisma/prisma/issues/3219) if there was an ability to override the types generated by the schema and give developers more flexibility on controlling the types.

**Comment by janpio** [comment]
(Might be worth creating a separate, explicit feature request issue for the idea "ability to override the types generated by the schema and give developers more flexibility on controlling the types" standalone)

**Comment by mmahalwy** [comment]
@janpio that's a good point. I will create a FR and tag you in it :)

**Comment by jkepps** [comment]
Did anything ever come of this? I am running into this exact use case

**Comment by blaykelarue** [comment]
adding an additional point here that having this ability would be greatly appreciated!

**Comment by leonardomgt** [comment]
I'm running into this exact scenario. It would be very nice to be able to do something like this in a PostgreSQL db:
```
model XTable {
  ...
  source      SourceType?  @db.String
}

enum SourceType {
  extension
  api
  import
}
```
or as @mmahalwy mentioned:
```
source  String  @cast(SourceType)
```

Atm, `@db.String` is not possible, so Prisma assumes the SourceType enum is an enum in the DB, which I don't want it to be. In the DB, I want source to be `source TEXT CHECK (source IN ('extension', 'api', 'import_list'))`.

Is there an alternative to achieve this behaviour?
